packages/create-ponder/src/helpers/graphqlTranslator.ts

```diff
@@ -0,0 +1,265 @@
+import { gql } from "graphql-tag";
+
+import type {
+  DefinitionNode,
+  NamedTypeNode,
+  NonNullTypeNode,
+} from "graphql/language";
+
+const GraphqlTypesToPonderTypes = {
+  ID: "string",
+  String: "string",
+  Int: "int",
+  BigInt: "bigint",
+  BigDecimal: "float",
+  Float: "float",
+  Boolean: "boolean",
+  Bytes: "hex",
+};
+
+export function translateSchema(schema: string): string {
+  const schemaAST = gql(schema);
+  const result: string[] = [];
+  const enums: string[] = [];
+  const enumsResult: string[] = [];
+  const idTypes: { [table: string]: string } = {};
+  const referenceFields: { [table: string]: { [key: string]: string } } = {};
+  const tables: {
+    [table: string]: {
+      columns: { [key: string]: string };
+      descriptions: { [key: string]: string };
+    };
+  } = {};
+  const joinedTables: { [name: string]: string } = {};
+
+  // Parse enums
+  // Need to do all the enums first before processing other types
+  // because there is no other way to determine if a graphql definition is referring to
+  // an enum or another table
+  Object.keys(schemaAST.definitions).forEach((definitionName) => {
+    const definition = schemaAST.definitions[
+      definitionName as any as number
+    ] as DefinitionNode;
+    if (definition.kind === "EnumTypeDefinition") {
+      const enumName = definition.name.value;
+      enums.push(enumName);
+      const values: string[] = [];
+      definition.values?.forEach((value) => {
+        values.push(value.name.value);
+      });
+
+      enumsResult.push(
+        `	${enumName}: p.createEnum(${JSON.stringify(values)}),\n\n`,
+      );
+    }
+  });
+
+  // Record id types for ObjectTypeDefinition
+  // This is needed for references later on
+  Object.keys(schemaAST.definitions).forEach((definitionName) => {
+    const definition = schemaAST.definitions[
+      definitionName as any as number
+    ] as DefinitionNode;
+    if (definition.kind === "ObjectTypeDefinition") {
+      const tableName = definition.name.value;
+      definition.fields?.forEach((field) => {
+        const fieldName = field.name.value;
+        if (fieldName === "id") {
+          const nonNullTypeNode = field.type as NonNullTypeNode;
+          const namedTypeNode = nonNullTypeNode.type as NamedTypeNode;
+          if (!namedTypeNode?.name?.value) {
+            console.log(field);
+            throw new Error(`Unsupported id type for ${tableName}`);
```

Comment:
Are you aware of a test case where this is reachable?

Reply:
Probably not. It's there as a check to see if the id type is not a list but yeah it probably won't happen on an already deployed subgraph.